### PR TITLE
Add PKCS#11 module in AppVeyor CI build

### PR DESCRIPTION
**Pull request by neusdan**
This PR:

- adds the external PKCS#11 header files into Botan
- adds a temporary fix for the PKCS#11 build and tests failures
- enables the PKCS#11 module in the AppVeyor CI build

Next steps will be to enable the PKCS#11 module in the other CI services and to add SoftHSMv2 and enable the PKCS#11 tests

### CODE REVIEW

**Comment by mouse07410** [comment]
As a temporary fix, why not instantiate PKCS11_RNG and seed HMAC_DRBG from it? I thought the main idea was to demonstrate that it can get randomness from PKCS11_RNG, not that HMAC_DRBG can be instantiated?

**Comment by neusdan** [comment]
Yeah we can do this, but as already mentioned the main idea is that the HMAC_DRBG seeds itself from   PKCS11_RNG automatically :-) Either way both "solutions" are temporary and I hope this gets fixed as soon as possible.

**Comment by mouse07410** [comment]
> Yeah we can do this, but as already mentioned the main idea is that the HMAC_DRBG seeds itself from PKCS11_RNG automatically :-)

I would much rather that it seeds itself from PKCS11_RNG manually when asked to, than to have a test in the PKCS11 branch that does nothing with PKCS11_<anything> at all. In that case might just drop HMAC_DRBG test altogether - as it is now it has nothing to do with PKCS11.

> Either way both "solutions" are temporary and I hope this gets fixed as soon as possible.

I concur. Though the "manual seeding" fix can be used as a workaround because it does get DRBG seeded from the token, and allows for reseeding at will. So I'd at least keep it as a sample for users who want to seed from the token *now*. While the fix is being contemplated (hopefully).

**Comment by neusdan** [comment]
done

**Comment by mouse07410** [comment]
Thank you!

File: src/scripts/ci/appveyor.yml [hunk 1]
```diff
@@ -23,9 +23,12 @@ install:
...
   - cl # check compiler version
+  
+  - appveyor DownloadFile http://download.qt.io/official_releases/jom/jom.zip -FileName jom.zip
+  - 7z e jom.zip
 
 build_script:
-  - python configure.py --cc=msvc --cpu=%PLATFORM% %MODE%
-  - nmake
+  - python configure.py --cc=msvc --cpu=%PLATFORM% %MODE% --with-pkcs11
+  - jom
```
Comment (randombit) [inline src/scripts/ci/appveyor.yml]: Why this change?

Comment (neusdan) [inline src/scripts/ci/appveyor.yml]: jom is a replacement for nmake which supports multi core  compilation and hopefully reduces compilation time.

Comment (webmaster128) [inline src/scripts/ci/appveyor.yml]: > reduces compilation time

Not without an explicit jobs parameter `/J` or `-j` ;)

Comment (neusdan) [inline src/scripts/ci/appveyor.yml]: no need to

> It automatically detects the number of processor cores in your PC by calling QThread::idealThreadCount()

Source: http://blog.qt.io/blog/2009/03/27/speeding-up-visual-c-qt-builds/

Comment (webmaster128) [inline src/scripts/ci/appveyor.yml]: Oh, yeah, idd. You are right. The information is still up to date in the current jom source. This is a strange default given that GNU make uses 1 job if not specified otherwise and jom should be a drop-in replacement for GNU make/nmake.

However, we experienced that in virtual environments like Travis, core detection functions like Linux `nproc` can detect some 32 cores while the power of the virtual machine is limited to 1.5 cores. Do we know what idealThreadCount() returns on AppVeyor? [We get 2 cores there](https://www.appveyor.com/docs/build-configuration#build-environment), so maybe we should fix that number.

Comment (neusdan) [inline src/scripts/ci/appveyor.yml]: Yeah agree, probably better to set it fix to 2.

**Comment by mouse07410** [comment]
Don't mean to be a pest, but when would this PR be merged?  And what's the timeframe for a more permanent fix (aka adding constructors to and changing behavior of HMAC_DRBG)?
